lib/Echidna/Transaction.hs

```diff
@@ -124,10 +124,14 @@ canShrinkTx _                                = True
 -- | Given a 'Transaction', generate a random \"smaller\" 'Transaction', preserving origin,
 -- destination, value, and call signature.
 shrinkTx :: MonadRandom m => Tx -> m Tx
-shrinkTx (Tx c s d g (C _ v) (C _ t, C _ b)) = let
+shrinkTx tx'@(Tx c _ _ _ (C _ v) (C _ t, C _ b)) = let
   c' = either (fmap Left . shrinkAbiCall) (fmap Right . pure) c
   lower x = w256 . fromIntegral <$> getRandomR (0 :: Integer, fromIntegral x) in
-    liftM5 Tx c' (pure s) (pure d) (pure g) (lower v) <*> fmap level (liftM2 (,) (lower t) (lower b))
+  sequence
```

Comment:
we can maybe rewrite this a bit cleaner: `uniform [set call =<< c', set value...] <*> pure tx'`

Reply:
I've cleaned this up a bit